staging/src/k8s.io/client-go/util/jsonpath/jsonpath.go

```diff
@@ -297,11 +297,11 @@ func (j *JSONPath) findFieldInValue(value *reflect.Value, node *FieldNode) (refl
 	var inlineValue *reflect.Value
 	for ix := 0; ix < t.NumField(); ix++ {
 		f := t.Field(ix)
-		jsonTag := f.Tag.Get("json")
-		parts := strings.Split(jsonTag, ",")
-		if len(parts) == 0 {
+		jsonTag, found := f.Tag.Lookup("json")
```

Comment:
Looking at this a bit more, I think it's only invalid to recurse if the outer struct has a field with the same name as the inner struct. Otherwise couldn't someone have intended to inline the inner field, and you'd be skipping that recursion? 

@lavalamp @mehdy In reality does this jsonpath parser only deal with types we control? Do we have a requirement that `json:".*,inline"` be used if you intend to inline a struct?